### ISSUE TITLE
Type validation through validateTypeOf

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#292]
```
The validation checks on declared return types do not appear to be working.  

For example:
    @Tainted @Untainted ExtendsAndAnnotation testMethod(){return null;}

Produces:

error: Found exception during TypeHierarchy.isSubtype of null and @Tainted 
@Untainted ExtendsAndAnnotation; invoke the compiler with -AprintErrorStack to 
see the stack trace.

But it should produce an type.invalid error. 

Or in the Flow Checker:

@Source(READ_SMS) @Sink({})  ExtendsAndAnnotation testMethod(){return null;}

No longer produces a forbidden flow warning. 


The call to validateTypeOf(ret) shown below does not issue a warning, because 
atypeFactory.getAnnotatedType(ret) does not return the explicitly written 
annotations.

// (BaseTypeVisitor, line ~401)
            Tree ret = node.getReturnType();
            if (ret != null &&
                    methodType.getReturnType().getKind() != TypeKind.VOID) {
                validateTypeOf(ret);
            }

Before revision e9436f0fc57e, this check was implemented using the 
typeValidator, which did produce the invalid type and forbidden flow warnings. 

typeValidator.visit(methodType.getReturnType(), node.getReturnType());


```

Original issue reported on code.google.com by `smil...@cs.washington.edu` on 18 Dec 2013 at 12:26
